app/discovery/provider/docker.go

```diff
@@ -272,3 +257,79 @@ func contains(e string, s []string) bool {
 	}
 	return false
 }
+
+type dockerClient struct {
+	client  http.Client
+	network string // network for IP selection
+}
+
+// NewDockerClient constructs docker client for given host and network
+func NewDockerClient(host, network string) DockerClient {
+	var schemaRegex = regexp.MustCompile("^(?:([a-z0-9]+)://)?(.*)$")
+	parts := schemaRegex.FindStringSubmatch(host)
+	proto, addr := parts[1], parts[2]
+	log.Printf("[DEBUG] Configuring docker client to talk to %s via %s", addr, proto)
```

Comment:
all other log messages don't capitalzie the text, we better to be consistent everywhere